pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -206,16 +207,25 @@ func (r *ReconcileNodeMaintenance) Reconcile(request reconcile.Request) (reconci
 		return r.reconcileAndError(instance, err)
 	}
 
-	if err = runCordonOrUncordon(r, node, true); err != nil {
+	if err = drain.RunCordonOrUncordon(r.drainer, node, true); err != nil {
 		return r.reconcileAndError(instance, err)
 	}
 
 	stop := make(chan struct{})
 	defer close(stop)
 
-	reqLogger.Infof("Evict all Pods from Node: %s", nodeName)
-	if err = drainPods(r, node, stop); err != nil {
-		return r.reconcileAndError(instance, err)
+	podsToDelete, errors := r.drainer.GetPodsForDeletion(nodeName)
+
+	if errors != nil && len(errors) != 0 {
+		aerror := utilerrors.NewAggregate(errors)
+		reqLogger.Errorf("Errors while listing pods for deletion: %v", aerror)
+	}
+
+	if len(podsToDelete.Pods()) != 0 {
```

Comment:
What is the purpose of this new functionality?

Reply:
This is done under initMaintenance for status purposes: 
https://github.com/kubevirt/node-maintenance-operator/blob/9fe31971cf36b090fbc6eee31f687c8cb30c3fdb/pkg/controller/nodemaintenance/nodemaintenance_controller.go#L286
And under reconcile and error:
https://github.com/kubevirt/node-maintenance-operator/blob/9fe31971cf36b090fbc6eee31f687c8cb30c3fdb/pkg/controller/nodemaintenance/nodemaintenance_controller.go#L311

Reply:
> we need to list the pods that are to be evicted prior to calling the DeleteOrEvictPods method of the upstream drain library.
> 217: GetPodsForDeletion - the call returns the list of pods to be evicted from the node
> 219-222: report errors if necessary. create a new aggregate error that includes the array of errors returned by GetPodsForDeletion
> 224: check if there is anything to evict prior to calling the evict call.

I understand what the new code is doing, but I'm yet to hear the reason why it is necessary.

Reply:
changed in later commit, please review.